sktime/transformations/series/tests/test_window_summarizer.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+"""Test extraction of features across (shifted) windows."""
+__author__ = ["Daniel Bartling"]
+
+import pandas as pd
+import pytest
+
+from sktime.datasets import load_airline
+from sktime.forecasting.model_selection import temporal_train_test_split
+from sktime.transformations.series.window_summarizer import LaggedWindowSummarizer
+
+# Load data that will be basis of tests
+y = load_airline()
+
+# Y Train will be univariate data set
+y_train, y_test = temporal_train_test_split(y)
+
+# Create Panel sample data
+mi = pd.MultiIndex.from_product([[0], y.index], names=["instances", "timepoints"])
+y_group1 = pd.DataFrame(y.values, index=mi, columns=["y"])
+
+mi = pd.MultiIndex.from_product([[1], y.index], names=["instances", "timepoints"])
+y_group2 = pd.DataFrame(y.values, index=mi, columns=["y"])
+
+y_grouped = pd.concat([y_group1, y_group2])
+
+# Create sample function dictionaries
+kwargs = {
+    "functions": {
+        "lag": ["lag", [[1, 1]]],
+        "mean": ["mean", [[1, 1], [5, 1]]],
+        "std": ["std", [[1, 2]]],
+    }
+}
+
+kwargs_alternames = {
+    "functions": {
+        "lag": ["lag", [[1, 1], [2, 1], [4, 1]]],
+        "covar": ["cov", [[1, 1], [2, 1], [4, 1]]],
+    }
+}
+
+kwargs_variant = {
+    "functions": {
+        "mean": ["mean", [[1, 7], [8, 7]]],
+        "covar_feature": ["cov", [[1, 28]]],
+    }
+}
+
+kwargs_empty = {"functions": {}}
+
+# Create transformer with kwargs
+transformer = LaggedWindowSummarizer(**kwargs)
+
+# Create transformer with different set of kwargs
+transformer_alternames = LaggedWindowSummarizer(**kwargs_alternames)
+
+# Create transformer with another different set of kwargs
+transformer_variant = LaggedWindowSummarizer(**kwargs_variant)
+
+# Create empty transformer
+transformer_empty = LaggedWindowSummarizer(**kwargs_empty)
+
+# Check if transformers works with non-Panel date
+Xt_univar = transformer.fit_transform(y_train)
```

Comment:
substantial functionality like this should happen *inside* a test function, or this may error out in test collection (not in execution)